control-plane/config-entries/controllersv2/gateway_class_config_controller.go

```diff
@@ -27,7 +27,8 @@ type GatewayClassConfigController struct {
 // +kubebuilder:rbac:groups=mesh.consul.hashicorp.com,resources=gatewayclassconfig/status,verbs=get;update;patch
 
 func (r *GatewayClassConfigController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	return r.Controller.ReconcileEntry(ctx, r, req, &meshv2beta1.GatewayClassConfig{})
+	//return r.MeshConfigController.ReconcileEntry(ctx, r, req, &meshv2beta1.GatewayClassConfig{})
+	return ctrl.Result{}, nil
```

Comment:
GatewayClassConfig was not registered with the consul API and was throwing errors.

Reply:
Let's merge this particular change separately so that the paper trail for the decision makes sense

https://github.com/hashicorp/consul-k8s/pull/3286